README.md

```diff
@@ -2,54 +2,123 @@
 
 # Terraform provider for Ironic
 
-This is a terraform provider that lets you provision baremetal servers managed by Ironic.
+## Provider declaration
 
-# Usage
-
-Example:
+Currently the provider only supports standalone noauth Ironic.  At a minimum, the Ironic
+endpoint URL must be specified. The user may also optionally specify an API microversion.
 
 ```terraform
 provider "ironic" {
-  "url" = "http://localhost:6385/v1"
-  "microversion" = "1.50"
+  url          = "http://localhost:6385/v1"
+  microversion = "1.50"
 }
+```
+
+## Resources
+
+This provider currently implements a number of native Ironic resources, described below. There is an additional resource
+called `ironic_deployment` that represents a deployment that consumes a baremetal node.
+
+## Nodes
 
+A node describes a hardware resource. If a node of the same name already exists, or
+`node_uuid` is specified, it's details would be updated. This could be used to provide
+BMC information for a discovered node.
+
+Nodes are distinct from the concept of a deployment, this resource manages the hardware-level configuration
+only. Destruction of a node object removes the node from Ironic.
+
+
+```
 resource "ironic_node_v1" "openshift-master-0" {
   name = "openshift-master-0"
-  target_provision_state = "active"
-  user_data = "${file("master.ign")}"
 
   ports = [
     {
-      "address" = "00:bb:4a:d0:5e:38"
-      "pxe_enabled" = "true"
-    }
+      pxe_enabled = true
+      address     = "00:bb:4a:d0:5e:38"
+    },
   ]
-
+  
   properties {
-    "local_gb" = "50"
-    "cpu_arch" =  "x86_64"
+    cpu_arch = "x86_64"
+    local_gb = 50
   }
 
-  instance_info = {
-    "image_source" = "http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2"
-    "image_checksum" = "26c53f3beca4e0b02e09d335257826fd"
-    "root_gb" = "25"
-    "root_device" = "/dev/vda"
+  root_device {
+   name = "/dev/vda"
   }
-
+   
   driver = "ipmi"
+  
   driver_info {
-			"ipmi_port"=      "6230"
-			"ipmi_username"=  "admin"
-			"ipmi_password"=  "password"
-			"ipmi_address"=   "192.168.111.1"
-			"deploy_kernel"=  "http://172.22.0.1/images/ironic-python-agent.kernel"
-			"deploy_ramdisk"= "http://172.22.0.1/images/ironic-python-agent.initramfs"
+    ipmi_port       =  "6230"
+    ipmi_username   =  "admin"
+    ipmi_password   =  "password"
+    ipmi_address    =  "192.168.111.1"
+    deploy_kernel   =  "http://172.22.0.1/images/ironic-python-agent.kernel"
+    deploy_ramdisk  =  "http://172.22.0.1/images/ironic-python-agent.initramfs"
   }
 }
 ```
 
+## Ports
+
+Ports may be specified as part of the node resource, or as a separate `ironic_port_v1`
+declaration.
+
+
+```
+resource "ironic_port_v1" "openshift-master-0-port-0" {
+  node_uuid   = "${ironic_node_v1.openshift-master-0.id}"
+  pxe_enabled = true
+  address     = "00:bb:4a:d0:5e:38"
+}
+```
+
+## Allocation
+
+```
+resource "ironic_allocation_v1" "openshift-master-allocation" {
+  count = 3
+
```

Comment:
nit: it may be handy to specify allocation name. or will it be derived from `openshift-master-allocation` in this example?